src/main/java/dibo/Storage.java

```diff
@@ -0,0 +1,123 @@
+package dibo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import dibo.exception.DiboException;
+import dibo.task.Deadline;
+import dibo.task.Event;
+import dibo.task.Task;
+import dibo.task.ToDo;
+
+/**
+ * The class to load data from and save data to the text file.
+ */
+public class Storage {
+    private static final DateTimeFormatter INPUT_FORMAT = DateTimeFormatter.ofPattern("MMM d yyyy");
+    private String filePath;
+
+    /**
+     * Constructs the Storage class.
+     *
+     * @param filePath The file path to the text file.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads the data from the text file and returns an ArrayList.
+     *
+     * @return An ArrayList to instantiate the TaskList object.
+     * @throws DiboException If an error occurs when loading.
+     */
+    public ArrayList<Task> loadData() throws DiboException {
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        try {
+            File f = new File(this.filePath);
+            Scanner sc = new Scanner(f);
+
+            while (sc.hasNextLine()) {
+                String taskDetails = sc.nextLine();
+                String[] details = taskDetails.split("\\|");
+
+                String type = details[0].trim();
+                Task task;
+                switch (type) {
+                case "todo":
+                    String description1 = details[2];
+                    task = new ToDo(description1);
+                    break;
+                case "deadline":
+                    String description2 = details[2];
+                    String by = details[3].trim();
+                    task = new Deadline(description2, convertToLocalDate(by));
+                    break;
+                case "event":
+                    String description3 = details[2];
```

Comment:
Not sure I like the name `description1`, perhaps you can rename it to `todoDescription` instead? Or have a singular `description` declaration at line 50 above which can be used for each of the cases since you have `description2` and `description3` as well?